lib/kubernetes-deploy/kubernetes_resource/base_set.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class BaseSet < KubernetesResource
+    def deploy_failed?
+      @pods.present? && @pods.any?(&:deploy_failed?)
+    end
+
+    def failure_message
+      @pods.map(&:failure_message).compact.uniq.join("\n")
+    end
+
+    def timeout_message
+      @pods.map(&:timeout_message).compact.uniq.join("\n")
+    end
+
+    def deploy_timed_out?
+      super || @pods.present? && @pods.any?(&:deploy_timed_out?)
+    end
+
+    def exists?
+      @found
+    end
+
+    def fetch_events
+      own_events = super
+      return own_events unless @pods.present?
+      most_useful_pod = @pods.find(&:deploy_failed?) || @pods.find(&:deploy_timed_out?) || @pods.first
+      own_events.merge(most_useful_pod.fetch_events)
+    end
+
+    def fetch_logs
+      container_names.each_with_object({}) do |container_name, container_logs|
```

Comment:
Add the `return {} unless @pods.present? # the kubectl command times out if no pods exist` guard from RS